### PR TITLE
Do not persist transient data in Enrollment

### PR DESCRIPTION
**Pull request by jasonaowen**
When migrating the Enrollment class to JPA annotations in b2ed85991988adab9f0858cd390cb7be4b9ade85, I missed that the `details` field did not have a corresponding entry in the legacy Hibernate mapping XML file. Mark it as `@Transient` so Hibernate does not complain about a missing column.

Issue #36 Use Hibernate 5, instead of 4
Pull request #83 Hibernate-5-ify Enrollment

### CODE REVIEW

**Comment by jasonaowen** [comment]
This isn't quite ready - I found a snag caused by this change.

**Comment by jasonaowen** [comment]
Okay, this should be ready to review!

**Comment by cecilia-donnelly** [comment]
Thanks @jasonaowen, I think I see what this does.  I see several other `getEm().merge` uses in `ProviderEnrollmentServiceBean.java` and I wonder if they need the same treatment?  It looks like all of them are updated before merging, though, so probably they're fine.

I also see that the ticket is merged a [couple](https://github.com/OpenTechStrategies/psm/blob/master/psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/BusinessProcessServiceBean.java#L242) [times](https://github.com/OpenTechStrategies/psm/blob/master/psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/BusinessProcessServiceBean.java#L545) in `BusinessProcessServiceBean.java`, but it looks like the ticket has already been updated in those cases, so: okay!

**Comment by cecilia-donnelly** [comment]
Also, I'm assuming I shouldn't be worried about the appearances of `note.setTicketId(0);`?  Those are just saying that this is a new `Ticket`/`Enrollment`?

**Comment by jasonaowen** [comment]
> I see several other getEm().merge uses in ProviderEnrollmentServiceBean.java and I wonder if they need the same treatment?

That is an excellent question, @cecilia-donnelly! I think you're right, there's [at least one more](https://github.com/OpenTechStrategies/psm/blob/7427a98842f127634206a91a31a358761f26287d/psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java#L2149-L2161) that needs to be updated. I will go fix. Thank you for catching that!

**Comment by jasonaowen** [comment]
> Also, I'm assuming I shouldn't be worried about the appearances of, e.g., `note.setTicketId(0);`? Those are just saying that this is a new `Ticket`/`Enrollment`?

I see two instances, on [line 1986](https://github.com/OpenTechStrategies/psm/blob/7427a98842f127634206a91a31a358761f26287d/psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java#L1986) and [line 2004](https://github.com/OpenTechStrategies/psm/blob/7427a98842f127634206a91a31a358761f26287d/psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java#L2004) is that what you're talking about? Those appear to me to be disassociating a service or note from an enrollment, rather than creating a new `Enrollment`.

In general, `thing.setId(0)` is the one that indicates a new `thing`; most all of the rest are just fields that have no particular meaning to Hibernate. That said, fields named things like `ticketId` or `profileId` suggest to me that there is a relationship there that neither Hibernate nor PostgreSQL know about, but should; I'm holding off on converting those for the moment, but fixing those kinds of things should be part of #57.

**Comment by jasonaowen** [comment]
@cecilia-donnelly okay, I'm going to talk through [the function that includes that second `merge`](https://github.com/OpenTechStrategies/psm/blob/7427a98842f127634206a91a31a358761f26287d/psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java#L2149-L2161), and explain why I think it works as-is - please check my logic!

- [L2149](https://github.com/OpenTechStrategies/psm/blob/7427a98842f127634206a91a31a358761f26287d/psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java#L2149) We're passed in `enrollment`
- [L2150](https://github.com/OpenTechStrategies/psm/blob/7427a98842f127634206a91a31a358761f26287d/psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java#L2150) We look up `ticket` based on `enrollment.ticketId`, and get a managed instance back that already has an ID set
- [L2151](https://github.com/OpenTechStrategies/psm/blob/7427a98842f127634206a91a31a358761f26287d/psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java#L2151) We get an unmanaged instance of `updatedProfile`
- [L2153-2156](https://github.com/OpenTechStrategies/psm/blob/7427a98842f127634206a91a31a358761f26287d/psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java#L2153-L2156) We delete the existing profile, if any
- [L2158](https://github.com/OpenTechStrategies/psm/blob/7427a98842f127634206a91a31a358761f26287d/psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java#L2158) We put the `updatedProfile` in the `@Transient` details
- [L2159](https://github.com/OpenTechStrategies/psm/blob/7427a98842f127634206a91a31a358761f26287d/psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java#L2159) We merge the managed instance... which should be a no-op? and I think always has been, since the only thing that changed was never persisted through the `Enrollment` entity.
- [L2160](https://github.com/OpenTechStrategies/psm/blob/7427a98842f127634206a91a31a358761f26287d/psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java#L2160) We use the helper method to persist the profile originally in `enrollment.details` and associate it with `enrollment.ticketId`.

So, I think that this is a very confusing function that could be much simpler, and it should be safe to leave it out of the scope of this PR.

@cecilia-donnelly:
> I guess stepping back: is this something that will come out in an error down the road?

This is also an excellent perspective, and from [the comment on the function](https://github.com/OpenTechStrategies/psm/blob/7427a98842f127634206a91a31a358761f26287d/psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java#L2144), we would see that error when approving a submitted enrollment. We can't even submit an enrollment yet, so in addition to the argument above that this might already be working correctly, yes, let's hold off on potential bugs until we can observe them more directly.

tl;dr I think we're good

**Comment by jasonaowen** [comment]
From irc:

> <@cdonnelly> jasonaowen: I'm fine with you merging it -- thanks for the ping
